extensions/libarchive/ReadArchiveStream.h

```diff
@@ -0,0 +1,90 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <utility>
+
+#include "io/OutputStream.h"
+#include "io/ArchiveStream.h"
+#include "core/logging/LoggerConfiguration.h"
+
+#include "archive_entry.h"
+#include "archive.h"
+
+namespace org::apache::nifi::minifi::io {
+
+class ReadArchiveStreamImpl : public ReadArchiveStream {
+  struct archive_ptr : public std::unique_ptr<struct archive, int(*)(struct archive*)> {
+    using Base = std::unique_ptr<struct archive, int(*)(struct archive*)>;
+    archive_ptr(): Base(nullptr, archive_read_free) {}
+    archive_ptr(std::nullptr_t): Base(nullptr, archive_read_free) {}
+    archive_ptr(struct archive* arch): Base(arch, archive_read_free) {}
+  };
```

Comment:
I think it's simpler to define deleter types and unique_ptr type aliases instead.

Reply:
don't we have to then manually specify the deleter `archive_read_free` on each construction? (specifically at `return nullptr;` in `createReadArchive`)

Reply:
done